mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerNoScan.java

```diff
@@ -647,6 +647,8 @@ public void testIcebergAndHmsTableProperties() throws Exception {
     Assert.assertEquals(HiveIcebergInputFormat.class.getName(), hmsTable.getSd().getInputFormat());
     Assert.assertEquals(HiveIcebergOutputFormat.class.getName(), hmsTable.getSd().getOutputFormat());
     Assert.assertEquals(HiveIcebergSerDe.class.getName(), hmsTable.getSd().getSerdeInfo().getSerializationLib());
+    // Add 1 second delay to reflect DDL_TIME change in HMS
+    Thread.sleep(1000);
```

Comment:
Sleeping for 1 second really adds up and makes tests slow. Instead, what about always setting the property to the last snapshot timestamp? That way you get more accurate information -- the timestamp of the last snapshot -- and you have something that you can also validate in a test using `Assert.equals`.

Reply:
> Sleeping for 1 second really adds up and makes tests slow... what about always setting the property to the last snapshot timestamp? 

I agree sleeping for 1 second is not ideal. Wasn't sure about a good alternative. Currently, the test runs fast enough that the table creation and subsequent snapshot write all happen in the same unix timestamp. So I can't write a test that shows the timestamp had changed, something like (table creation timestamp < write timestamp)